openpype/hosts/maya/plugins/publish/extract_look.py

```diff
@@ -255,6 +255,14 @@ def process(self, instance):
         hashes = results["fileHashes"]
         remap = results["attrRemap"]
 
+        # Import Reference if the option is enabled
+        ref_import = instance.data.get("importReference", True)
+        if ref_import:
+            reference_node = cmds.ls(type="reference")
+            for r in reference_node:
+                rFile = cmds.referenceQuery(r, f=True)
+                cmds.file(rFile, importReference=True)
```

Comment:
Sorry. Why would I want to import into my workfile during publish? And why on look publishing. This seems odd.

The [resulting published look already does NOT preserve references due to this argument](https://github.com/pypeclub/OpenPype/blob/8ae92351a7e3e1758cca4a62e4170749ed93a914/openpype/hosts/maya/plugins/publish/extract_look.py#L284) and thus results in a file without references as if they were imported.

Reply:
I think this want meant for rendering not look publish, 

We should have the possibility to import all references for published render workfile which is used for rendering. (it  should not affect the workfile itself)

- to prevent render farm issues with referenced files
- to help with cloud rendering and dealing with dependencies